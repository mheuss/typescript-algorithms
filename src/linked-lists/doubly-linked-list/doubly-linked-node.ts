export class DoublyLinkedNode<T> {
  public previous: DoublyLinkedNode<T> | null = null;
  public next: DoublyLinkedNode<T> | null = null;
  public payload: T;

  constructor(payload: T, previous?: DoublyLinkedNode<T>) {
    this.payload = payload;
    if (previous) {
      previous.next = this;
      this.previous = previous;
    }
  }
}
