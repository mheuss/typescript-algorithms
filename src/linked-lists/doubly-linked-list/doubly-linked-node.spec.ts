import { DoublyLinkedNode } from './doubly-linked-node';

describe('List Node unit test', () => {
  it('Should allow for creation of numerous node items', () => {
    const node = new DoublyLinkedNode<number>(5);
    const secondNode = new DoublyLinkedNode<number>(15, node);
    expect(node).toMatchSnapshot();
    expect(secondNode).toMatchSnapshot();
  });
});
