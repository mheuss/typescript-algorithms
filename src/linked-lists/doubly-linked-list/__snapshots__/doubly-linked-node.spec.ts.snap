// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Node unit test Should allow for creation of numerous node items 1`] = `
DoublyLinkedNode {
  "next": DoublyLinkedNode {
    "next": null,
    "payload": 15,
    "previous": [Circular],
  },
  "payload": 5,
  "previous": null,
}
`;

exports[`List Node unit test Should allow for creation of numerous node items 2`] = `
DoublyLinkedNode {
  "next": null,
  "payload": 15,
  "previous": DoublyLinkedNode {
    "next": [Circular],
    "payload": 5,
    "previous": null,
  },
}
`;
